### DISCUSSION TITLE
ReadMe file showing up instead of React app, plus build deployment issues

### DISCUSSION THREAD

**Discussion by dcho2001** [community/community#138210] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi, I am a beginner programmer learning React.

I am having trouble getting my site to appear at the url https://dcho2001.github.io/Little-Lemon-Web-App/. Instead of my React app, I see a React ReadMe file. I read that you can go to the Pages tab in Settings and change the source to index.html , something to do with changing the branch and folder, but as I look there are limited options for me to do so. Here are some screenshots below:

<img width="778" alt="Screenshot 2024-09-09 at 1 28 09 PM" src="https://github.com/user-attachments/assets/f90ff946-ed85-496c-81cf-79d3f79f8fa9">
<img width="782" alt="Screenshot 2024-09-09 at 1 28 05 PM" src="https://github.com/user-attachments/assets/20b78397-0351-416e-896a-b8694141693c">

I asked Co-Pilot and ChatGPT for help, and they told me I should push my files to a gh-pages branch. I did a bunch of things like adding the gh-pages to my package.json, configuring my homepage, setting up a GitHub actions workflow, and pushed it to the repository, but it still doesn't let me select gh-pages as a branch.

Also, I see there is a failure to deploy my build, as seen below:

<img width="898" alt="Screenshot 2024-09-09 at 1 36 10 PM" src="https://github.com/user-attachments/assets/15008114-06a1-41c8-8c85-0ae865507a2a">

Here is my repository:
https://github.com/dcho2001/Little-Lemon-Web-App

I've been stuck on this the past couple days, so if someone could at least steer me in the right direction that would be helpful, thanks!

**Comment by ThisIs-Developer** [comment]
### GitHub Pages Deployment for React App:

1. **Add `homepage` in `package.json`:**
   ```json
   "homepage": "https://dcho2001.github.io/Little-Lemon-Web-App/"
   ```

2. **Install `gh-pages`:**
   ```bash
   npm install --save gh-pages
   ```

3. **Add Deploy Scripts to `package.json`:**
   ```json
   "scripts": {
     "predeploy": "npm run build",
     "deploy": "gh-pages -d build"
   }
   ```

4. **Deploy to `gh-pages` Branch:**
   Run:
   ```bash
   npm run deploy
   ```

5. **Set GitHub Pages Branch:**
   Go to **Settings > Pages**, select `gh-pages` branch, folder set to `/ (root)`.

6. **Check for `.nojekyll`:**
   Ensure the `.nojekyll` file is present in the `build` folder.

7. **Check Build Logs:**
   Inspect any errors in the deployment logs.

8. **Clear Browser Cache:**
   After deployment, clear cache or use incognito mode to view the latest version.

For more detailed guidance, you can go through this article: [Deploy React App to Github Pages](https://www.dhiwise.com/post/simplify-deployment-how-to-deploy-react-app-to-github-page)

If this solves your problem, **mark the answer as helpful 😪**!

**Reply by dcho2001** [reply]
.nojekyll file?

**Reply by ThisIs-Developer** [reply]
The `.nojekyll` file tells GitHub Pages **not** to treat your site as a Jekyll project. It prevents GitHub from ignoring files like those starting with `_`, which can cause issues when deploying React apps. It's usually auto-generated when using `gh-pages`, but you can manually add it to your `build` folder if needed.

**Reply by ThisIs-Developer** [reply]
@dcho2001 I guess your problem is solved, but if not, please let me know! Otherwise, go ahead and mark it as the answer and close the discussion 😄👍!

**Comment by Milda100** [comment]
"Heads up to anyone using Vite instead of create-react-app. Vite creates a "dist" folder instead of "deploy" on build so  you need to add "deploy": "gh-pages -d dist"   instead of "deploy": "gh-pages -d deploy" when adding to scripts. If you don't do this Vite won't locate the "build" folder and the gh-pages branch will not get created."

I read this comment on youtube and it fixed everything for me.
